src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -342,32 +459,74 @@ private JButton getNewButton() {
 			String State = InventoryStatus.draft.toString();
 			String user = UserBrowsingManager.getCurrentUser();
 			int checkResults = 0;
-			if (inventoryRowSearchList == null || inventoryRowSearchList.size() < 1) {
-				MessageDialog.error(null, "angal.inventory.noproduct.msg");
+			if (inventoryRowSearchList == null || inventoryRowSearchList.isEmpty()) {
+				MessageDialog.error(null, "angal.inventory.cannotsaveinventorywithoutproducts.msg");
 				return;
 			}
 			LocalDateTime now = LocalDateTime.now();
 			if (dateInventory.isAfter(now)) {
 				MessageDialog.error(null, "angal.inventory.notdateinfuture.msg");
-				return;
+				return ;
 			}
-
+			if (!lotDeletes.isEmpty() || !inventoryRowsToDelete.isEmpty()) {
+				for (Map.Entry<Integer, Lot> entry : lotDeletes.entrySet()) {
+					MedicalInventoryRow invRow = medicalInventoryRowManager.getMedicalInventoryRowById(entry.getKey());
+					if (invRow != null) {
+						invRow.setLot(null);
+						try {
+							medicalInventoryRowManager.updateMedicalInventoryRow(invRow);
+							movStockInsertingManager.deleteLot(entry.getValue());
+						} catch (OHServiceException e) {
+							OHServiceExceptionUtil.showMessages(e);
+							return ;
+						}
+					}
+					
+		        }
+				try {
+					medicalInventoryRowManager.deleteMedicalInventoryRows(inventoryRowsToDelete);
+				} catch (OHServiceException e) {
+					OHServiceExceptionUtil.showMessages(e);
+					return ;
+				}
+			}
+			
 			if ((inventory == null) && (mode.equals("new"))) {
-				String reference = referenceTextField.getText().trim();
-				if (reference.equals("")) {
+				newReference = referenceTextField.getText().trim();
+				if (newReference.equals("")) {
 					MessageDialog.error(null, "angal.inventory.mustenterareference.msg");
-					return;
+					return ;
 				}
-				if (medicalInventoryManager.referenceExists(reference)) {
+				if (medicalInventoryManager.referenceExists(newReference)) {
 					MessageDialog.error(null, "angal.inventory.referencealreadyused.msg");
-					return;
+					return ;
 				}
 				inventory = new MedicalInventory();
-				inventory.setInventoryReference(reference);
+				inventory.setInventoryReference(newReference);
 				inventory.setInventoryDate(dateInventory);
 				inventory.setStatus(State);
 				inventory.setUser(user);
 				inventory.setInventoryType(InventoryType.main.toString());
+				if (chargeType != null) {
+					inventory.setChangeType(chargeType.getCode());
+				} else {
+					inventory.setChangeType(null);
+				}
+				if (dischargeType != null) {
+					inventory.setDischangeType(dischargeType.getCode());
+				} else {
+					inventory.setDischangeType(null);
+				}
+				if (supplier != null) {
+					inventory.setSupplier(supplier.getSupId());
+				} else {
+					inventory.setSupplier(null);
+				}
+				if (destination != null) {
+					inventory.setDestination(destination.getCode());
+				} else {
+					inventory.setDestination(null);
+				}
 				MedicalInventory meInventory;
```

Comment:
And all other occurences

```suggestion
				MedicalInventory newInventory;
```